<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>520</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_8">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_28">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>舵机</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_30">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_10">
          <item>
           <widget class="QGroupBox" name="groupBox_3">
            <property name="title">
             <string>IO口配置</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_12" stretch="5,1">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_5">
                 <item>
                  <layout class="QVBoxLayout" name="verticalLayout_2">
                   <item>
                    <widget class="QLabel" name="label_2">
                     <property name="text">
                      <string>左上</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="SerLeftUp_IO"/>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <layout class="QVBoxLayout" name="verticalLayout">
                   <item>
                    <widget class="QLabel" name="label">
                     <property name="text">
                      <string>左下</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="SerLeftDown_IO"/>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_6">
                 <item>
                  <layout class="QVBoxLayout" name="verticalLayout_3">
                   <item>
                    <widget class="QLabel" name="label_3">
                     <property name="text">
                      <string>右上</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="SerRightUp_IO"/>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <layout class="QVBoxLayout" name="verticalLayout_4">
                   <item>
                    <widget class="QLabel" name="label_4">
                     <property name="text">
                      <string>右下</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="SerRightDown_IO"/>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="UpdateSers_IO">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Ignored" vsizetype="Minimum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>更新IO口</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_5">
            <property name="title">
             <string>参数调整</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_10">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,3">
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_7">
                 <item>
                  <widget class="QLabel" name="label_5">
                   <property name="text">
                    <string>选择舵机</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QComboBox" name="ParaChooseSer">
                   <item>
                    <property name="text">
                     <string>请选择</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>左上</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>左下</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>右上</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>右下</string>
                    </property>
                   </item>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_8">
                 <item>
                  <widget class="QLabel" name="label_6">
                   <property name="text">
                    <string>输入旋转参数</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QDoubleSpinBox" name="ParaSerRotate">
                   <property name="minimum">
                    <double>-1.000000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>1.000000000000000</double>
                   </property>
                   <property name="singleStep">
                    <double>0.050000000000000</double>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,3">
               <item>
                <widget class="QPushButton" name="ParaSerGo">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Ignored" vsizetype="Expanding">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>执
行</string>
                 </property>
                 <property name="iconSize">
                  <size>
                   <width>10</width>
                   <height>10</height>
                  </size>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_9">
                 <property name="spacing">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QPushButton" name="ParaSerSave2Up">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string>保存为上升参数</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="ParaSerSave2Reset">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string>保存为待机参数</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="ParaSerSave2Down">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string>保存为下落参数</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_6">
          <property name="title">
           <string>舵机控制</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="4,3">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>选择舵机</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="CtrlChooseSer">
               <item>
                <property name="text">
                 <string>请选择</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>左上</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>左下</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>右上</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>右下</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>全部</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <property name="spacing">
              <number>3</number>
             </property>
             <item>
              <widget class="QPushButton" name="CtrlSerUp">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>上升</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="CtrlSerReset">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>待机</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="CtrlSerDown">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>下落</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_8">
       <property name="title">
        <string>其他设置</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_27">
        <item>
         <widget class="QPushButton" name="pushButton_17">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>占位</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_18">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>占位</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_19">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>占位</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_22">
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>超声波传感器</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_29">
        <item>
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string>IO口配置</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_26" stretch="4,1">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_13">
               <item>
                <widget class="QLabel" name="label_8">
                 <property name="text">
                  <string>左上</string>
                 </property>
                 <property name="textFormat">
                  <enum>Qt::AutoText</enum>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignCenter</set>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_11">
                 <item>
                  <widget class="QLabel" name="label_9">
                   <property name="text">
                    <string>Trig</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="TrigLeftUp_IO"/>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_10">
                   <property name="text">
                    <string>Echo</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="EchoLeftUp_IO"/>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_14">
               <item>
                <widget class="QLabel" name="label_11">
                 <property name="text">
                  <string>左下</string>
                 </property>
                 <property name="textFormat">
                  <enum>Qt::AutoText</enum>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignCenter</set>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_15">
                 <item>
                  <widget class="QLabel" name="label_12">
                   <property name="text">
                    <string>Trig</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="TrigLeftDown_IO"/>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_13">
                   <property name="text">
                    <string>Echo</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="EchoLeftDown_IO"/>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_16">
               <item>
                <widget class="QLabel" name="label_14">
                 <property name="text">
                  <string>右上</string>
                 </property>
                 <property name="textFormat">
                  <enum>Qt::AutoText</enum>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignCenter</set>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_17">
                 <item>
                  <widget class="QLabel" name="label_15">
                   <property name="text">
                    <string>Trig</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="TrigRightUp_IO"/>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_16">
                   <property name="text">
                    <string>Echo</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="EchoRightUp_IO"/>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_18">
               <item>
                <widget class="QLabel" name="label_17">
                 <property name="text">
                  <string>右下</string>
                 </property>
                 <property name="textFormat">
                  <enum>Qt::AutoText</enum>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignCenter</set>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_19">
                 <item>
                  <widget class="QLabel" name="label_18">
                   <property name="text">
                    <string>Trig</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="TrigRightDown_IO"/>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_19">
                   <property name="text">
                    <string>Echo</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="EchoRightDown_IO"/>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="UpdateHCSR_IO">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Ignored" vsizetype="Minimum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>更新IO口</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_7">
          <property name="title">
           <string>查看探测距离</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_25">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_20">
               <item>
                <widget class="QLabel" name="label_20">
                 <property name="text">
                  <string>选择传感器</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="ChooseHCSR">
                 <item>
                  <property name="text">
                   <string>请选择</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>左上</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>左下</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>右上</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>右下</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_21">
               <item>
                <widget class="QLabel" name="label_21">
                 <property name="text">
                  <string>返回值</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="DisReturn">
                 <property name="text">
                  <string>探测中...</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_24">
             <item>
              <widget class="QPushButton" name="HCSRGo">
               <property name="text">
                <string>开始探测</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="FlagEmptyDis">
               <property name="text">
                <string>标记空桶距离</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="FlagFullDis">
               <property name="text">
                <string>标记满载距离</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="quitBtn">
       <property name="text">
        <string>退出</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
